res/schema.sql

```diff
@@ -21,5 +23,56 @@ CREATE TABLE IF NOT EXISTS exchange_rate_history (
     value FLOAT
 );
 
-
 CREATE INDEX IF NOT EXISTS exchange_rate_history_date_index ON exchange_rate_history (date);
+
+DO $$ BEGIN
+    CREATE TYPE transaction_type AS ENUM ('deposit', 'withdraw', 'buy', 'sell');
+EXCEPTION
+    WHEN duplicate_object THEN null;
+END $$;
+
+CREATE TABLE IF NOT EXISTS transaction_history (
+    uuid TEXT PRIMARY KEY,
+    timestamp TIMESTAMP WITH TIME ZONE NOT NULL,
+    user_uuid TEXT NOT NULL REFERENCES users (uuid),
+    type transaction_type NOT NULL,
+    amount FLOAT NOT NULL,
+    total_after_transaction FLOAT NOT NULL
```

Comment:
Shouldn't we separate this to `total_btc_after_transaction` and `total_usd_after_transaction`?

Reply:
Currently each transaction affects either only BTC or only USD. This is what I was talking about in #41, e.g. `buy` transaction also causes a `withdraw` transaction.

Reply:
How about when we buy btc (so amount of usd decreases) - is it processed as two separate transactions?

Reply:
In my opinion, we should keep both balances after transaction for USD and BTC. Mainly the reason is that on buy query not only BTC balance changes, but USD as well. We also have to remember that this trigger will execute twice per buy/sell operation, because we use two queries.

So on buy operation it would add "buy" and then "withdraw" to transaction history. Imo it is not ideal. The problem is, the database can't know whether we perform "buy" or only "withdraw", so there are two solutions:
- somehow merge buy queries into one, use a trigger (which would be the best, if possible) and label both operations as single "buy"/"sell";
- don't use trigger and perform update to this table from backend as well (yes, it would require adjusting all wallet endpoints)

Reply:
Done (I went with the first approach)